### PR TITLE
Add DeGate origin

### PR DESCRIPTION
**Pull request by deep-quality-dev**
### description of pull request:

- Add DeGate origin

### reference
https://github.com/degatedev/degate-sdk-golang
https://api-docs.degate.com/spot/#24hr-ticker-price-change-statistics

### CODE REVIEW

File: pkg/datapoint/origin/degate.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	// LastQty            string `json:"last_qty"`
+	// BidPrice           string `json:"bid_price"`
+	// BidQty             string `json:"bid_qty"`
+	// AskPrice           string `json:"ask_price"`
+	// AskQty             string `json:"ask_qty"`
+	// OpenPrice          string `json:"open_price"`
+	// HighPrice          string `json:"high_price"`
+	// LowPrice           string `json:"low_price"`
+	Volume string `json:"volume"`
+	// QuoteVolume     string `json:"quote_volume"`
```
Comment (MDobak) [inline pkg/datapoint/origin/degate.go:99]: If this is not used, just remove it.

File: pkg/datapoint/origin/degate.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+			return nil, err
+		}
+	}
+
+	for _, pair := range pairs {
+		index := d.pairs.byPair(pair)
+		if index < 0 {
+			points[pair] = datapoint.Point{Error: fmt.Errorf("unsupported pair: %s", pair.String())}
+			continue
+		}
```
Comment (MDobak) [inline pkg/datapoint/origin/degate.go:186]: What's the purpose of this check? Why there's a need to provide a list of supported pairs in the config struct?

**Review by MDobak — CHANGES_REQUESTED** [review]

File: pkg/datapoint/origin/degate.go (hunk 2 above)
Comment (MDobak) [reply pkg/datapoint/origin/degate.go:149]: > I think specifying the supported pairs helps us to understand which pairs use DeGate, vise versa. Wdyt?

We're not doing it for any other origin, and we always conduct research before making a pricing model, so I don't think these kinds of checks would ever be useful.

Comment (deep-quality-dev) [reply pkg/datapoint/origin/degate.go]: Good catch, removed pairs in this [commit](https://github.com/chronicleprotocol/oracle-suite/pull/301/commits/50aa526d83c1d96f6231c012c80df795116b56d9).

**Pull request closed without merging**